api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -31,9 +31,24 @@ message ConfigDump {
   // * *clusters*: :ref:`ClustersConfigDump <envoy_api_msg_admin.v2alpha.ClustersConfigDump>`
   // * *listeners*: :ref:`ListenersConfigDump <envoy_api_msg_admin.v2alpha.ListenersConfigDump>`
   // * *routes*:  :ref:`RoutesConfigDump <envoy_api_msg_admin.v2alpha.RoutesConfigDump>`
+  //
+  // Dump a specific configuration component with
+  // :ref:`/config_dump?key={} <operations_admin_interface_config_dump_by_key>` You can see
```

Comment:
Did you consider generalizing this to allow arbitrary proto path lookups? What I'm thinking is that internally Envoy is basically outputing a proto with the config dump. We could conceivably specify an arbitrary path through this proto to find the subtree we want to output. This would take the form of a list like [top level key, field 3, item 2 in list] etc. Another way to think of this is as some sort of xpath for proto. This mechanism would be generalizable to any proto output from admin endpoint.

Maybe this is overkill, but just wanted to check on whether we may want something more sophisticated than top-level key and a specialization to config dump here.

Reply:
That's interesting. The other extension I was considering was allowing users to pass a name and then searching through the dump for protos with a name field matching that value. I think I prefer the approach you're describing as it is more generic.

I'll poke around and see how one might use passed strings to access fields of the proto.

Reply:
I think this may do what we want https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.util.field_mask_util

Reply:
@spenceral yep, exactly, `FieldMask` makes sense here IMHO.